### DISCUSSION TITLE
question(usage): Stylus inside HTML

### DISCUSSION THREAD

**Discussion by Kristinita** [stylus/stylus#2614] [Q&A]
### 1. Summary

Is it possible to use Stylus inside HTML files? If no, are there plans to do this?

### 2. Example in other languages

For CoffeeScript:

+ [**Live demo on Codepen**](https://codepen.io/Kristinita/pen/aXZJEZ):

```html
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8"/>
        <title>
            GCSE file
        </title>
        <script type="text/javascript" src="https://cdn.rawgit.com/jashkenas/coffeescript/1.10.0/extras/coffee-script.js"></script>
        <script type="text/coffeescript">
                do ->
                    cx = '013024336414733191742:sps98skj394'
                    gcse = document.createElement('script')
                    gcse.type = 'text/javascript'
                    gcse.async = true
                    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx
                    s = document.getElementsByTagName('script')[0]
                    s.parentNode.insertBefore gcse, s
                    return
            </script>
    </head>
    <body>
        <!-- HTML5 validation, see:
                https://productforums.google.com/d/msg/customsearch/XE-cs3_oVuY/ohwVpwEUAgAJ and https://developers.google.com/custom-search/docs/element#html5 -->
        <div class="gcse-search" data-enableHistory="true" data-enableAutoComplete="true" data-autoCompleteMaxCompletions="10" data-enableOrderBy="false">
        </div>
    </body>
</html>
```

I can paste CoffeeScript code inside HTML. Is it possible similar behavior for Stylus?

### 3. Argumentation

I understand, that this behavior may be not best for real projects.

But, for example, it would be nice for [**Stack Exchange snippets**](https://stackoverflow.blog/2014/09/16/introducing-runnable-javascript-css-and-html-code-snippets/). [**Preprocessors doesn't support at the time**](https://meta.stackexchange.com/questions/297810/scss-sass-and-other-css-preprocessor-support-for-code-snippets), users needs to use native CSS in examples. But I don't want native CSS; Stylus — is the best CSS preprocessor of the world :smile: .

### 4. Not helped

I try some Google queries → I don't find, how I can get expected behavior.

![Stylus Google 1](https://i.imgur.com/vVUYfhK.png)

![Stylus Google 2](https://i.imgur.com/gqW62Al.png)

Thanks.

**Comment by vendethiel** [comment]
Stylus' website has a "try it online" page. Have you looked at how it works?

**Comment by ghost** [comment]
You can't run both of them on browser if that is the question. 

https://stackoverflow.com/questions/7860942/is-it-possible-to-compile-coffeescript-code-in-script-tags-in-html-files

**Comment by MoussaAdam** [comment]
@atilkan there is a little misunderstanding here, he is not asking "can i compile stylus code in html **before serving it**", he want to compile stylus on the browser of the client, instead of pre-processing it, and the answer is yes as stated by @vendethiel .
its just a transpiler written in javascript, browsers can run javascript code,so its possible and [less.js](http://lesscss.org/) is an example of that.

so @kaime yes its possible, but i don't recommend that
- it will make your website slower
- it will not be so easy to implement
- stylus is not designed with browser compatibility in mind
